repository/src/raw.rs

```diff
@@ -86,25 +95,20 @@ pub trait RawRepository: Send + Sync {
     // Commit-related methods
     // ----------------------
 
-    /// Create a commit with the branch.
-    ///
-    /// Same as `checkout_clean() && git checkout <branch> && git add -A && git commit`.
+    /// Create a commit from the currently checked out branch.
     async fn create_commit(
         &mut self,
-        branch: &Branch,
         commit_message: &str,
         diff: Option<&str>,
-    ) -> Result<(), Error>;
+    ) -> Result<CommitHash, Error>;
 
-    /// Create a commit with the given list to overwrite files.
-    ///
-    /// * `files`: the list of `(file_path, file_content_to_overwrite)`.
-    async fn create_commit_overwrite(
-        &mut self,
-        branch: &Branch,
-        commit_message: &str,
-        files: Vec<(String, String)>,
-    ) -> Result<(), Error>;
+    /// Create a semantic commit from the currently checked out branch.
+    async fn create_semantic_commit(&mut self, commit: SemanticCommit)
```

Comment:
How commit can be created from semantic_commit? What needs to contain semantic_commit's content in the new commit?

Reply:
@JeongHunP Have you checked out `to_semantic_commit` function in this PR? Maybe you're looking for this.

Reply:
@TomTaehoonKim I checked but I didn't understand it. So is it right that those title and body in SemanticCommit should be converted to commit message as ASO(simple-git example) shows?